closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -460,6 +519,67 @@ int decryptFinal(byte[] in, int inOfs, int len,
         return ret;
     }
 
+
+    /**
+     * Performs decryption operation for the last time.
+     *
+     * @param src the input buffer with the data to be decrypted
+     * @param dst the offset in <code>in</code>
+     * @return the number of bytes placed into the <code>out</code> buffer
+     */
+    int decryptFinal(ByteBuffer src, ByteBuffer dst)
+        throws IllegalBlockSizeException, AEADBadTagException,
+        ShortBufferException {
+
+        // Get array from source
+        byte[] src_arr = getBbArray(src);
+        int src_ofs = src.hasArray() ? (src.position() + src.arrayOffset()) : 0;
+
+        // Get array from destination
+        byte[] dst_arr;
+        int dst_offset;
+        if (dst.hasArray()) {
+            dst_arr = dst.array();
+            dst_offset = dst.position() + dst.arrayOffset();
+        } else {
+            dst_arr = new byte[dst.capacity()];
+            dst_offset = 0;
+        }
+
+        int len = decryptFinal(src_arr, src_ofs, src.remaining(), dst_arr, dst_offset);
+        
+        // Advance source buffer position
+        src.position(src.limit());
+        
+        // Update destination buffer
+        if (dst.hasArray()) {
+            dst.position(dst.position() + len);
+        } else {
+            dst.put(dst_arr, 0, len);       
+        }
+        
+        return len;
+    }
+
+    /**
+     * Gets the byte array behind a buffer.
+     * Tries to use ByteBuffer.array(). If this is not available, the function uses ByteBuffer.get()
+     *
+     * @param src the buffer whose byte[] is needed
+     * @return the byte array with the buffer's content
+     */
+    private static byte[] getBbArray(ByteBuffer src) {
+        byte[] arr;
+        if (src.hasArray()) {
+            arr = src.array();
+        } else {
+            ByteBuffer cpy = src.duplicate();
+            arr = new byte[cpy.remaining()];
+            if (arr.length > 0) cpy.get(arr, 0, arr.length);
```

Comment:
Please use braces (`{}`) with the body on a separate line:
```
if (arr.length > 0) {
    cpy.get(arr, 0, arr.length);
}
```

Reply:
Fixed.